src/main.c

```diff
@@ -146,6 +160,8 @@ static int cbc_essiv_iv_derivation(uint32_t sector_number, uint8_t * iv,
     sector_number_buff[1] = (big_endian_sector_number >> 8) & 0xff;
     sector_number_buff[2] = (big_endian_sector_number >> 16) & 0xff;
     sector_number_buff[3] = (big_endian_sector_number >> 24) & 0xff;
+    /* Copy bytes of the SD serial hash */
+    memcpy(sector_number_buff+4, sd_serial_digest, sizeof(sector_number_buff)-4);
```

Comment:
Hi,

Here, sizeof(sector_number_buff) - 4 is smaller than sd_serial_digest (SHA256_DIGEST_SIZE). Although there is no size comparison check and the +4 / -4 is hard to understand (it took me some time to understand that it was the separation between the counter part and the digest part of the sector_number_buff)

Is it possible to add it ?